src/main/java/Duke.java

```diff
@@ -1,10 +1,78 @@
+import Exceptions.DukeException;
+import java.util.Scanner;
 public class Duke {
-    public static void main(String[] args) {
-        String logo = " ____        _        \n"
-                + "|  _ \\ _   _| | _____ \n"
-                + "| | | | | | | |/ / _ \\\n"
-                + "| |_| | |_| |   <  __/\n"
-                + "|____/ \\__,_|_|\\_\\___|\n";
-        System.out.println("Hello from\n" + logo);
-    }
-}
+    private final Ui ui;
+    private TaskList taskList;
+    private final Storage storage;
+    public static final String DATA_FILE_PATH = "data/duke.txt";
+
+    public Duke(String filepath) {
+        this.ui = new Ui();
+        this.storage = new Storage(filepath);
+        try {
+            this.taskList = new TaskList(this.storage.load());
+        } catch (DukeException e) {
+            this.ui.showDukeError(e);
+            this.taskList = new TaskList();
+        }
+    }
+
+    /**
+     * Loads the list of tasks from the data file.
+     * @return The status of loading the list of tasks.
+     */
+    public String loadTasks() {
+        try {
+            this.taskList = new TaskList(this.storage.load());
+            return "Tasks loaded successfully!";
+        } catch (DukeException e) {
+            this.taskList = new TaskList();
+            return "Tasks failed to load" + this.ui.showDukeError(e);
+        } catch (Exception e) {
+            return this.ui.showException(e);
+        }
+    }
+
+    /**
+     * Shows the welcome message.
+     * @return the welcome message.
+     */
+    public String showWelcome() {
+        return this.ui.showWelcome();
+    }
+
+    /**
+     * Returns Chewy's response to the user's input.
+     * @param userInput The user's input.
+     * @return Chewy's response.
+     */
+    public String getResponse(String userInput) {
+        try {
+            if (userInput.equals("bye")) {
```

Comment:
Maybe it would be better to use switch statements here instead of verbose if else statements?